clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -63,6 +75,51 @@ hello-world-foo"
     );
 }
 
+#[test]
+fn suggest_hidden_long_flag_aliases() {
+    let mut cmd = Command::new("exhaustive")
+        .arg(
+            clap::Arg::new("hello-world")
+                .long("hello-world")
+                .visible_alias("hello-world-foo")
+                .alias("hidden-world"),
+        )
+        .arg(
+            clap::Arg::new("hello-moon")
+                .long("hello-moon")
+                .visible_alias("hello-moon-foo")
+                .alias("hidden-moon")
+                .hide(true),
+        )
+        .arg(
+            clap::Arg::new("goodbye-world")
+                .long("goodbye-world")
+                .alias("goodbye-world-foo"),
+        );
```

Comment:
In your tests cases, could you use words that hint at the test case as its easier to tell whats going on?

For example, here you use `-foo` suffix when you could do either
- `-visible`
- `-hidden`

Reply:
(put another way, I should be able to have a good idea of what is going on with the next commit's diff with minimal looking at the test setup)

Reply:
One test was updated but we should update both

This was solved enough for whether aliases are visible but doesn't address whether the arg is visibule

Reply:
This is the only remaining task